packages/Search/examples/bvh_driver/bvh_driver.cpp

```diff
@@ -99,145 +55,230 @@ int main_( Teuchos::CommandLineProcessor &clp, int argc, char *argv[] )
-        // radius chosen in order to control the number of results per query
-        // NOTE: minus "1+sqrt(3)/2 \approx 1.37" matches the size of the boxes
-        // inserted into the tree (mid-point between half-edge and
-        // half-diagonal)
-        double const r = 2. * std::cbrt( static_cast<double>( n_neighbors ) *
-                                         3. / ( 4. * M_PI ) ) -
-                         ( 1. + std::sqrt( 3. ) ) / 2.;
-        Kokkos::parallel_for(
-            "bvh_driver:setup_radius_search_queries",
-            Kokkos::RangePolicy<ExecutionSpace>( 0, n_queries ),
-            KOKKOS_LAMBDA( int i ) {
-                queries( i ) = DataTransferKit::within( random_points( i ), r );
-            } );
-        Kokkos::fence();
+template <class DeviceType>
+void BM_radius_search( benchmark::State &state )
+{
+    int const n_values = state.range( 0 );
+    int const n_queries = state.range( 1 );
+    int const n_neighbors = state.range( 2 );
+    int const buffer_size = state.range( 3 );
+
+    DataTransferKit::BVH<DeviceType> bvh(
+        constructBoxes<DeviceType>( n_values ) );
+    auto const queries =
+        makeSpatialQueries<DeviceType>( n_values, n_queries, n_neighbors );
 
+    for ( auto _ : state )
+    {
         Kokkos::View<int *, DeviceType> offset( "offset" );
         Kokkos::View<int *, DeviceType> indices( "indices" );
-
-        start = std::chrono::high_resolution_clock::now();
+        auto const start = std::chrono::high_resolution_clock::now();
         bvh.query( queries, indices, offset, buffer_size );
-        end = std::chrono::high_resolution_clock::now();
+        auto const end = std::chrono::high_resolution_clock::now();
+        std::chrono::duration<double> elapsed_seconds = end - start;
+        state.SetIterationTime( elapsed_seconds.count() );
+    }
+}
 
-        elapsed_seconds = end - start;
-        os << "radius " << elapsed_seconds.count() << "\n";
+class KokkosScopeGuard
+{
+  public:
+    KokkosScopeGuard( int &argc, char *argv[] )
+    {
+        Kokkos::initialize( argc, argv );
     }
+    ~KokkosScopeGuard() { Kokkos::finalize(); }
+};
 
-    return 0;
-}
+#define REGISTER_BENCHMARK( DeviceType )                                       \
+    BENCHMARK_TEMPLATE( BM_construction, DeviceType )                          \
+        ->Arg( n_values )                                                      \
+        ->UseManualTime()                                                      \
+        ->Unit( benchmark::kMicrosecond );                                     \
+    BENCHMARK_TEMPLATE( BM_knn_search, DeviceType )                            \
+        ->Args( {n_values, n_queries, n_neighbors} )                           \
+        ->UseManualTime()                                                      \
+        ->Unit( benchmark::kMicrosecond );                                     \
+    BENCHMARK_TEMPLATE( BM_radius_search, DeviceType )                         \
```

Comment:
So this lets you parameter sweep through the benchmark?

Reply:
You can specify range for arguments.  See google/benchmark doc.